src/main/java/com/synopsys/integration/detect/tool/detector/inspectors/nuget/DotNetRuntimeAvailabilityVerifier.java

```diff
@@ -0,0 +1,63 @@
+package com.synopsys.integration.detect.tool.detector.inspectors.nuget;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.detectable.detectable.exception.DetectableException;
+
+public class DotNetRuntimeAvailabilityVerifier {
+    private static final String DOTNET_RUNTIME_PATTERN_VERSION_PLACEHOLDER = "<DOTNET_VERSION>";
+    private static final String DOTNET_RUNTIME_PATTERN_WITHOUT_VERSION = "Microsoft\\.(AspNetCore|NETCore){1}\\.(All|App){1}(\\s){1}"
```

Comment:
This seems like a very complicated Regex that will match small finite list of Strings. Perhaps a more simple lookup and replace is appropriate.

Reply:
Yeah, I talked to @ekerwin about regex usage usage yesterday. I will be updating the logic here to be more universally maintainable.